table/block_based_table_reader.cc

```diff
@@ -813,7 +813,10 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
       CachableEntry<IndexReader> index_entry;
       unique_ptr<InternalIterator> iter(
           new_table->NewIndexIterator(ReadOptions(), nullptr, &index_entry));
-      index_entry.value->CacheDependencies(pin);
+      assert(index_entry.value != nullptr || !iter->status().ok());
+      if (index_entry.value != nullptr) {
+        index_entry.value->CacheDependencies(pin);
+      }
```

Comment:
Can we log here why creating the index failed? It is not clear to me how it could happen in non-failure scenarios.

Reply:
That's what the assert intends to catch. We want to make sure it only fails to create index in failure scenarios, i.e., when the iterator contains a failure status. The iterator's failure status gets propagated to the caller, who logs it.